src/afrolabs/components/kafka.clj

```diff
@@ -624,6 +644,66 @@
         [_ cfg]
       (assoc cfg ConsumerConfig/CLIENT_ID_CONFIG client-id))))
 
+;; Once the consumer loads, fetches the endOffsets. Once the consumer reaches those offsets
+;; will send this value to the channel. This is distinct from `CaughtUpNotifications` in that
+;; the latter only fires when the consumer is up-to-date, whereas this strategy fires
+;; once the consumer reaches the offsets that were current when it started up.
+(defstrategy CaughtUpOnceNotifications
+  [& chs]
+  (let [caught-up-ch (csp/chan)
+        caught-up-mult (csp/mult caught-up-ch)
+        end-offsets-at-start (atom nil)]
+
+    ;; tap all of the provided channels into the mult/ch that we notify on
+    (doseq [ch chs] (csp/tap caught-up-mult ch))
+
+    (reify
+      IShutdownHook
+      (shutdown-hook
+          [_ _]
+        (csp/close! caught-up-ch))
+
+      IConsumerPostInitHook
+      (post-init-hook
+          [_ consumer]
+        ;; Test if we've caught up to the last offset for every topic-partition we're consuming from
+        (let [topic-partition-assignment (.assignment ^KafkaConsumer consumer)
+
+              end-offsets
+              (into #{}
+                    (map (fn [[tp o]] {:topic (.topic ^TopicPartition tp)
+                                       :partition (.partition ^TopicPartition tp)
+                                       :offset o}))
+                    (.endOffsets ^KafkaConsumer consumer
+                                 topic-partition-assignment))]
+          (reset! end-offsets-at-start end-offsets)))
```

Comment:
We store the end-offsets that are "current" right after the consumer is initialized.